cosmos-common/src/main/scala/com/mesosphere/cosmos/render/PackageDefinitionRenderer.scala

```
@@ -46,11 +50,11 @@ object PackageDefinitionRenderer {
     pkgDef: universe.v4.model.PackageDefinition,
     options: Option[JsonObject],
     marathonAppId: Option[AppId]
-  ): Either[PackageDefinitionRenderError, JsonObject] = {
-    pkgDef.marathon.map { marathon =>
+  ): Option[JsonObject] = {
+    pkgDef.marathon.flatMap { marathon =>
       val defaultOptionsAndUserOptions: JsonObject = mergeDefaultAndUserOptions(pkgDef, options)
 
-      validateOptionsAgainstSchema(pkgDef, defaultOptionsAndUserOptions).flatMap { _ =>
+      if (validateOptionsAgainstSchema(pkgDef, defaultOptionsAndUserOptions)) {
```

I noticed that, since you're now throwing exceptions instead of returning an `Either` from `validateOptionsAgainstSchema()`, you made the return type `Boolean`. However, `false` is never returned from that method! So you should be able to change it to return `Unit` instead, and then call it directly here without needing to wrap it in an `if`.

Ultimately, you should be able to change the `flatMap` into a `map` on the first line of the method (more comments about this later):

```scala
pkgDef.marathon.map { marathon =>
  // ...
}
```